src/misc.test.ts

```diff
@@ -0,0 +1,111 @@
+import {
+  isNonEmptyArray,
+  isNullOrUndefined,
+  isObject,
+  hasProperty,
+  RuntimeObject,
+} from './misc';
+
+describe('miscellaneous', () => {
+  describe('isNonEmptyArray', () => {
+    it('identifies non-empty arrays', () => {
+      [[1], [1, 2], [1, 2, 3, 4, 5]].forEach((nonEmptyArray) => {
+        expect(isNonEmptyArray<unknown>(nonEmptyArray)).toBe(true);
+      });
+    });
+
+    it('identifies empty arrays', () => {
+      expect(isNonEmptyArray<unknown>([])).toBe(false);
+    });
+  });
+
+  describe('isNullOrUndefined', () => {
+    it('identifies null and undefined', () => {
+      expect(isNullOrUndefined(null)).toBe(true);
+      expect(isNullOrUndefined(undefined)).toBe(true);
+    });
+
+    it('identifies non-nullish values', () => {
+      [false, 1, 0, -1, '', [], () => undefined].forEach(
+        (nonNullOrUndefinedValue) => {
+          expect(isNullOrUndefined(nonNullOrUndefinedValue)).toBe(false);
+        },
+      );
+    });
+  });
+
+  describe('isObject', () => {
+    class Foo {}
```

Comment:
[nit] What do you think about defining this at the top of the file so it stands out?

Reply:
I think it's better to keep it close to where it's used, particularly since it's only used in this one spot. What do you think?

Reply:
Okay. If another `describe` block also needs to define a Foo class, that class will conflict with this one, but I guess doing so will produce a linting error, so it's probably okay.